apps/openmw/mwmechanics/aiactivate.cpp

```diff
@@ -23,33 +23,29 @@ namespace MWMechanics
         return new AiActivate(*this);
     }
 
-    bool AiActivate::execute (const MWWorld::Ptr& actor, CharacterController& characterController, AiState& state, float duration)
+    bool AiActivate::execute(const MWWorld::Ptr& actor, CharacterController& characterController, AiState& state, float duration)
     {
-        ESM::Position pos = actor.getRefData().getPosition(); //position of the actor
         const MWWorld::Ptr target = MWBase::Environment::get().getWorld()->searchPtr(mObjectId, false); //The target to follow
 
         actor.getClass().getCreatureStats(actor).setDrawState(DrawState_Nothing);
 
-        if(target == MWWorld::Ptr() ||
-            !target.getRefData().getCount() || !target.getRefData().isEnabled()  // Really we should be checking whether the target is currently registered
+        if (target == MWWorld::Ptr() ||
+            !target.getRefData().getCount() || !target.getRefData().isEnabled()  // Really we should check whether the target is currently registered
                                                                                 // with the MechanicsManager
-                )
-            return true;   //Target doesn't exist
+            )
+        return true;   //Target doesn't exist
 
-        //Set the target desition from the actor
+        //Set the target destination for the actor
         ESM::Pathgrid::Point dest = target.getRefData().getPosition().pos;
 
-        if(distance(dest, pos.pos[0], pos.pos[1], pos.pos[2]) < 200) { //Stop when you get close
-            actor.getClass().getMovementSettings(actor).mPosition[1] = 0;
+        if (pathTo(actor, dest, duration, 200)) //Go to the destination
+        {
+            // activate when reached

```

Comment:
Thx for the note. This was exactly the reason of the merge conflict resolved in f57858b.